test/Sentry.Tests/SpanTracerTests.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using NSubstitute;
+using Xunit;
+
+namespace Sentry.Tests
+{
+    public class SpanTracerTests
+    {
+        [Fact]
+        public async Task SetExtra_DataInserted_NoDataLoss()
+        {
+            // Run 20 times to avoid flacky tests scapping.
+            for (int amount = 0; amount < 20; amount++)
+            {
+                // Arrange
+                var hub = Substitute.For<IHub>();
+                var transaction = new SpanTracer(hub, null, null, SentryId.Empty, "");
+
+                // Act
+                var tasks = Enumerable.Range(1, 4).Select((_) => Task.Run(() =>
```

Comment:
There's no guarantee of parallelism here because there's no synchronization.
It's possible/likely that the tasks will activate in series and their callback run also in series.

You need to synchronize (with the snippet I shared earlier) to block them all before the code you're attempting to test is ready to be executed by each task to get closer to parallelism here.

Have them all WaitOne before the for loop and unblock them all at once from the test thread once they are all ready for it.